src/main/java/oracle/goldengate/delivery/handler/marklogic/listprocesor/WriteListProcessor.java

```diff
@@ -199,7 +206,10 @@ public void process(List<WriteListItem> writeListItems) {
         headers.put("importDate", OffsetDateTime.now().format(DateTimeFormatter.ISO_OFFSET_DATE_TIME));
         headers.put("schema", item.getSourceSchema());
         headers.put("table", item.getSourceTable());
-        headers.put("operation", item.getOperation());
+
+        if (!handlerProperties.getTransformName().equals("mlRunIngest") ){
```

Comment:
`handlerProperties.getTransformName()` could return null, right? If so, this could throw an NPE. What is the actual logic that is needed here? Only add the operation header if there is a transform that is not the DHF ingest one?

Reply:
To avoid operation header to be added as it isn't allowed on the DHF5.2